pkg/manifests/config.go

```diff
@@ -467,6 +476,35 @@ func (c *Config) NoProxy() string {
 	return c.ClusterMonitoringConfiguration.HTTPConfig.NoProxy
 }
 
+func (c *Config) LoadEnforcedBodySizeLimit(pcr PodsCapacityReader, ctx context.Context) error {
+	enforcedBodySizeLimit := c.ClusterMonitoringConfiguration.PrometheusK8sConfig.EnforcedBodySizeLimit
+	if enforcedBodySizeLimit == "" {
+		podCapacity, err := pcr.PodsCapacity(ctx)
+		if err != nil {
+			return fmt.Errorf("error fetching pod capacity: %v", err)
+		}
+		enforcedBodySizeLimit = calculateBodySizeLimit(podCapacity)
+	}
+	if err := pm.ValidateSizeField(enforcedBodySizeLimit); err != nil {
+		return err
+	}
+	c.ClusterMonitoringConfiguration.PrometheusK8sConfig.EnforcedBodySizeLimit = enforcedBodySizeLimit
+	return nil
+}
+
+func calculateBodySizeLimit(podCapacity int) string {
+	const samplePerPod = 50
+	const sizePerSample = 200 // 200 Bytes
+
+	bodySize := podCapacity * samplePerPod * sizePerSample
+	if bodySize < minimalSizeLimit {
+		bodySize = minimalSizeLimit
+		klog.Infof("Calculated scrape body size limit is too small, using default value %v instead", minimalSizeLimit)
+	}
+
+	return fmt.Sprintf("%dKB", int(bodySize/1024))
```

Comment:
Lets take another safety margin and round up the value to the next MB.

```suggestion
	return fmt.Sprintf("%dMB", math.Ceil(float64(bodySize)/(1024*1024))))
```